packages/api-derive/src/chain/util.ts

```diff
@@ -24,34 +31,32 @@ export function createBlockNumberDerive <T extends { number: Compact<BlockNumber
 }
 
 export function getAuthorDetails (header: Header, queryAt: QueryableStorage<'rxjs'>): Observable<[Header, Vec<AccountId> | null, AccountId | null]> {
-  // this is Moonbeam specific
-  if (queryAt.authorMapping && queryAt.authorMapping.mappingWithDeposit) {
-    const mapId = header.digest.logs[0] && (
-      (header.digest.logs[0].isConsensus && header.digest.logs[0].asConsensus[1]) ||
-      (header.digest.logs[0].isPreRuntime && header.digest.logs[0].asPreRuntime[1])
-    );
-
-    if (mapId) {
-      return combineLatest([
-        of(header),
-        queryAt.session
-          ? queryAt.session.validators()
-          : of(null),
-        queryAt.authorMapping.mappingWithDeposit<IOption<{ account: AccountId } & Codec>>(mapId).pipe(
-          map((opt) =>
-            opt.unwrapOr({ account: null }).account
+  // nimbus consensus stores authorship in the header logs
+  const { logs: [log] } = header.digest;
+  const loggedAuthor = (log && (
+    (log.isConsensus && log.asConsensus[0].isNimbus && log.asConsensus[1]) ||
+    (log.isPreRuntime && log.asPreRuntime[0].isNimbus && log.asPreRuntime[1])
+  )) || null;
+
+  const validators = (queryAt.session)
+    ? queryAt.session.validators()
+    : null;
+
+  const mappedAuthor = (loggedAuthor)
+    // use the author mapping pallet if available (ie: moonbeam, moonriver)
+    ? (queryAt.authorMapping && queryAt.authorMapping.mappingWithDeposit)
+      ? queryAt.authorMapping.mappingWithDeposit<IOption<{ account: AccountId } & Codec>>(loggedAuthor)
+        .pipe(map((opt) => opt.unwrapOr({ account: null }).account))
+      // use the session pallet if available (ie: manta, calamari)
+      : (queryAt.session && queryAt.session.queuedKeys)
+        ? queryAt.session.queuedKeys<Observable<Vec<ITuple<[AccountId32, INimbusSessionKeys]>>>>()
+          .pipe(map((o: Observable<Vec<ITuple<[AccountId32, INimbusSessionKeys]>>>) => o.pipe(
+            map((qk: ITuple<[AccountId32, INimbusSessionKeys]>[]) =>
+              (qk.find(([_, { nimbus }]) => nimbus.toHex() === loggedAuthor.toHex()) || [null])[0]
+            )))
           )
-        )
-      ]);
-    }
-  }
-
-  // normal operation, non-mapping
-  return combineLatest([
-    of(header),
-    queryAt.session
-      ? queryAt.session.validators()
-      : of(null),
-    of(null)
-  ]);
+        : null
+    : null;
+
+  return combineLatest([of(header), of(validators), of(mappedAuthor)]) as Observable<[Header, Vec<AccountId> | null, AccountId | null]>;
```

Comment:
This cast should not be needed - `combineLatest` does actually return the correct types.

The `of(validators)` here seems suspect since it could be an `Observable<AccountId>` or `null` (as opposed to `Observable<null>` as per lines 41 to 43 above). 

So with this is seems to be `Observable<Observable<AccountId>>` or `Observable<null>` since we apply `of` on it in combineLatest.